pkg/sinit/writer.go

```diff
@@ -0,0 +1,21 @@
+package sinit
+
+import (
+	"errors"
+	"log"
+)
+
+type logLoggerWriter struct {
+	dest *log.Logger
+}
+
+var ErrLoggerNotInitialized = errors.New("log logger is not initialized")
+
+func (w *logLoggerWriter) Write(p []byte) (n int, err error) {
+	if w.dest == nil {
+		return 0, ErrLoggerNotInitialized
```

Comment:
I personally would add a tiny test for this check (as it prevents a crash if this ever gets changed/refactored), should be absolutely trivial.